spec/enumerables_spec.rb

```diff
@@ -0,0 +1,273 @@
+        expect(range.my_none? { |ele| ele > 5 }).to be true
+      end
+
+      it 'should return false if any of the collection items evaluates to true' do
+        expect(array5.my_none? { |ele| ele == 3 }).to be false
+        expect(range.my_none? { |ele| ele > 4 }).to be false
+      end
+    end
+
+    context 'when pattern is given as argument' do
+      it 'should return true if pattern doesn\'t match any element in collection' do
+        expect(array3.my_none?(String)).to be true
+        expect(array2.my_none?(/s/)).to be true
+      end
+
+      it 'should return false if pattern === any element in collection' do
+        expect(array3.my_none?(nil)).to be false
+        expect(array2.my_none?(/o/)).to be false
+      end
+    end
+  end
+
+  describe '#my_count' do
+    context 'when block given' do
+      it 'counts number of collection items yielding a true value' do
+        expect(array5.my_count { |ele| ele == 2 }).to eq(3)
+        expect(array1.my_count { |ele| ele > 2 }).to eq(2)
+      end
+    end
+
+    context 'when no block is given' do
+      it 'should return number of items in collection' do
+        expect(array2.count).to eq(4)
+        expect(array4.count).to eq(4)
+      end
+    end
+
+    context 'when argument is given' do
+      it 'should return number of items in collection that are equal to argument' do
+        expect(array5.my_count(2)).to eq(3)
+      end
+    end
+  end
+
+  describe '#my_map' do
+    context 'when block given' do
+      it 'should return a new array with results of running block once for every collection item' do
+        expect(array1.my_map { |ele| ele + 1 }).to eq([2, 3, 4, 5])
+        expect(range.my_map { |ele| ele + 2 }).to eq([3, 4, 5, 6, 7])
+        expect(range.my_map { 'cat' }).to eq(%w[cat cat cat cat cat])
+      end
+    end
+
+    context 'when no block is given' do
+      it 'should return an Enumerator ' do
+        expect(array1.my_map).to be_a_kind_of(Enumerator)
+        expect(array1.my_map.to_a).to eq(array1)
+      end
+    end
+  end
+
+  describe '#my_inject' do
+    context 'when block is given' do
+      context 'when no initial value is specified' do
+        it 'should combine all collection items by applying a binary operation' do
+          expect(range.my_inject { |acc, num| acc + num }).to eq(15)
+          expect(array1.my_inject { |acc, num| acc + num }).to eq(10)
+          expect(array1.my_inject { |acc, num| acc * num }).to eq(24)
+          expect(array2.my_inject { |acc, num| acc + num }).to eq('twothreetentwelve')
+        end
+      end
+
+      context 'when initial value is specified' do
+        it 'should combine all collection items with binary operation starting with specified initial value' do
+          expect(range.my_inject(2) { |acc, num| acc + num }).to eq(17)
+          expect(array1.my_inject(2) { |acc, num| acc + num }).to eq(12)
+          expect(array1.my_inject(5) { |acc, num| acc * num }).to eq(120)
```

Comment:
* [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))